client/src/common/types/schema.ts

```diff
@@ -0,0 +1,58 @@
+type Category = number | string | boolean;
+
+export interface AnnotationColumnSchema {
+  categories?: Category[];
+  name: string;
+  type: "string" | "float32" | "int32" | "categorical" | "boolean";
+  writable: boolean;
+}
+
+interface DataFrameSchema {
+  nObs: number;
+  nVar: number;
+  // TODO(thuang): Not sure what other types are available
+  type: "float32";
+}
+
+export interface LayoutColumnSchema {
```

Comment:
Similarly, this would more appropriately be called `EmbeddingSchema`, as it does not describe a single dataframe column, and the word `layout` was another historically and very incorrect name.   Today we universally call these embeddings, not layouts.